contracts/contracts/levels/DoubleEntryPoint.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+interface DelegateERC20 {
+  function delegateTransfer(address to, uint256 value, address origSender) external returns (bool);
+}
+
+interface IAgent {
+    function handleTransaction(address user) external returns(bool);
```

Comment:
I'd remove the bool return value and add msg.data as explained in previous comment.

```suggestion
    function handleTransaction(address user, bytes calldata msgData) external;
```